src/data/bmg_plate.py

```diff
@@ -28,26 +31,72 @@ def __init__(self, barcode: str, plate_array: np.ndarray) -> None:
         :param filepath: path to the file consisting plate
         """
         self.barcode = barcode
-        self.plate_array = plate_array
-        self.control_statistics()
-        self.z_factor = 1 - (
-            3 * (self.std_pos + self.std_neg) / (self.mean_neg - self.mean_pos)
+        self.plate_array = plate_array.astype(np.float32)
+        self.pos = self.plate_array[:, -1]
+        self.neg = self.plate_array[:, -2]
+        (
+            self.std_pos,
+            self.std_neg,
+            self.mean_pos,
+            self.mean_neg,
+            self.z_factor,
+        ) = self.control_statistics(self.pos, self.neg)
+        self.pos_wo, self.outliers_pos = self.find_outliers(
+            self.pos, self.mean_pos, self.std_pos
         )
-
-    def control_statistics(self) -> None:
+        self.neg_wo, self.outliers_neg = self.find_outliers(
+            self.neg, self.mean_neg, self.std_neg
+        )
+        (
+            self.std_pos_wo,
+            self.std_neg_wo,
+            self.mean_pos_wo,
+            self.mean_neg_wo,
+            self.z_factor_wo,
+        ) = self.control_statistics(self.pos_wo, self.neg_wo)
+
+    def control_statistics(self, pos: np.ndarray, neg: np.ndarray) -> tuple:
         """
         Calculate statistic for control values e.g. two last columns of a plate
```

Comment:
considering `control_statistics` does not use any `Plate` parameters it can be extracted as a standalone function

+docs: missing params in docstring